front/lib/api/auth_wrappers.ts

```diff
@@ -276,7 +276,9 @@ export function withPublicAPIAuthentication<T, U extends boolean>(
           req,
           res,
           auth,
-          null as U extends true ? Authenticator : null
+          (opts.allowUserOutsideCurrentWorkspace
```

Comment:
Can you explain the diff not sure I understand it (and it looks quite critical 😬)

Reply:
This one is specific to the runApp* endpoints (POST `/runs/index`), as it's the only endpoint which use `allowUserOutsideCurrentWorkspace: true`. This endpoint expects 2 `auth` objects (and require both of them) - one based on the workspace in the url (usually dust-apps), the other one based on the workspace of the key. In the case of oauth token, the token is not tied to a workspace, just a user, who can be in multiple workspaces.
Actually using the same dust-apps workspace is clearly not a good idea, as we'll get credentials from dust-apps, which is not something we want. Better revert this - it just means this endpoint will not be usable with oauth tokens. Thanks for the catch.

Reply:
Why not have a proper mechanism that similar to key that creates a workspaceAuth and a tokenAuth ?

Reply:
The only workspace we have is from the request path, which allows us to create the "workspaceAuth". As opposed to api key, we can't get a workspace from an oauth token as it does not contain any workspace information. Best we could do would be to take randomly one workspace of the user (for most users, they only have one) to create a "tokenAuth".

Reply:
I see :+1: